### PR TITLE
Support regx exclude interface list

### PR DESCRIPTION
**Pull request by XuHandsome**
### fix: To solve character encoding error in driver version return
some network interface return driver version
Maybe the country of manufacturer, the character "\x00" in driver version response, it leads the metrics pages context-Type change to "binary", like this:
```bash
transceiver_driver_version_info{driver_version="1.0.0\\x00-50-generic",interface="enp0s5"} 1
transceiver_driver_version_info{driver_version="1.0\\x00.0-50-generic",interface="cali2d8e0dfad7a"} 1
transceiver_driver_version_info{driver_version="1.0\\x00.0-50-generic",interface="calib073cd807d5"} 1
transceiver_driver_version_info{driver_version="1.0\\x00.0-50-generic",interface="calidf4d15675a8"} 1
transceiver_driver_version_info{driver_version="1.0\\x00.0-50-generic",interface="calif0538e521e7"} 1
transceiver_driver_version_info{driver_version="2.3\\x00.0-50-generic",interface="docker0"} 1
transceiver_driver_version_info{driver_version="5.15.0-50-generic",interface="kube-ipvs0"} 1
```

### feat: support regx exclude interface list
In virtualized operating system, like installed docker proxmox and other, so much virtual bridge device, like `docker0` `docker1` `bond0` `bond1` `vmbr0v5050` `vmbr1v3000` and other, so i added the regx support, just with the `-exclude.interfaces` parameter, same as before.
```bash
./transceiver-exporter -exclude.interfaces bond*,vmbr*,tap*,docker*,eth0
```

### CODE REVIEW

**Comment by vidister** [comment]
Hi, thanks for contributing. Looks good to me, I will run some tests, add some documentation and create a release then.

**Comment by johannwagner** [comment]
Actually, your proposed syntax is not regex but some type of Linux globs but your proposed code does actual regex matching. I would also like to introduce a feature switch to enable Regex support to ensure a non-breaking new version.

File: main.go [hunk 1]
```diff
@@ -20,7 +20,7 @@ var (
 	listenAddress            = flag.String("web.listen-address", "[::]:9458", "Address to listen on")
 	metricsPath              = flag.String("web.telemetry-path", "/metrics", "Path under which to expose metrics")
 	collectInterfaceFeatures = flag.Bool("collector.interface-features.enable", true, "Collect interface features")
-	excludeInterfaces        = flag.String("exclude.interfaces", "", "Comma seperated list of interfaces to exclude")
+	excludeInterfaces        = flag.String("exclude.interfaces", "", "Comma seperated list of interfaces regular to exclude")
```
Comment (johannwagner) [inline main.go:23]: Please introduce a seperate feature switch to enable Regex matching

File: transceiver-collector/util.go [hunk 2]
```diff
@@ -21,3 +26,9 @@ func boolToFloat64(b bool) float64 {
 func milliwattsToDbm(mw float64) float64 {
 	return 10 * math.Log10(mw)
 }
+
+func quoteToascii(str string) string {
```
Comment (johannwagner) [inline transceiver-collector/util.go:30]: `quoteToASCII`

File: main.go [hunk 3]
```diff
@@ -83,9 +84,12 @@ func (t transceiverCollectorWrapper) Describe(ch chan<- *prometheus.Desc) {
 func handleMetricsRequest(w http.ResponseWriter, request *http.Request) {
 	registry := prometheus.NewRegistry()
 
-	excludedIfaceNames := strings.Split(*excludeInterfaces, ",")
-	for index, excludedIfaceName := range excludedIfaceNames {
-		excludedIfaceNames[index] = strings.Trim(excludedIfaceName, " ")
+	var excludedIfaceNames []string
+	if len(*excludeInterfaces) != 0 {
```
Comment (johannwagner) [inline main.go:88]: Since we got new features, it would also be nice to support regex for included interfaces.

**Review by johannwagner — CHANGES_REQUESTED** [review]

**Comment by vidister** [comment]
superseded by #9 

nevertheless, thank you for your contribution.

**Pull request closed without merging**